src/main/java/edition/academy/seventh/serivce/EmpikScrapping.java

```diff
@@ -0,0 +1,89 @@
+package edition.academy.seventh.serivce;
+
+import edition.academy.seventh.database.model.Book;
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Phaser;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.select.Elements;
+
+/**
+ * Scrap data from empik.com bookstore website in sales section using JSoup library.
+ *
+ * @author Bartosz Kupajski
+ */
+public class EmpikScrapping implements IPromotionScrapping {
+
+    private List<Book> listOfBooks = new CopyOnWriteArrayList<>();
+    private ExecutorService service = Executors.newFixedThreadPool(40);
+    private Phaser phaser = new Phaser(1);
+  /**
+        Above that number SSLException is thrown
+     */
+    private final int maxConnections = 1201;
+
+  /**
+   * Index in for loop is incremented by 30 because of the fact that
+   * every page URL of empik sales section contains id witch is incremented by 30.
+   *
+   * @return list of books after all threads finish their jobs
+   *
+   */
+  @Override
+  public List<Book> scrapPromotion() {
+
+      for (int i = 1; i <= maxConnections; i = i + 30) {
+        service.submit(createScrappingTask(i));
+        }
+
+        phaser.arriveAndAwaitAdvance();
+      return listOfBooks;
+    }
+
+  private Runnable createScrappingTask(int serchSiteNumber) {
```

Comment:
serchSiteNumber??? co to jest? jeśli chodzi o numer przeszukiwanej strony to powinno być numberOfSearchedSite